src/Type.zig

```diff
@@ -417,6 +454,26 @@ pub fn isUnsignedInt(ty: Type, comp: *Compilation) bool {
     };
 }
 
+pub fn getArray(ty: *const Type) ?*const Type {
+    return switch (ty.specifier) {
+        .array, .static_array, .incomplete_array, .decayed_array, .decayed_static_array, .decayed_incomplete_array => ty,
+        .typeof_type => ty.data.sub_type.getArray(),
+        .typeof_expr => ty.data.expr.ty.getArray(),
+        .attributed => ty.data.attributed.base.getArray(),
+        else => null,
+    };
+}
+
+pub fn isEnum(ty: Type) bool {
```

Comment:
This function isn't needed; you can just use `ty.is(.@"enum")`

Reply:
Yes, it ultimately calls into `Type.get` which should handle typeof- and attributed types

Reply:
fixed.